mobile_verifier/src/boosting_oracles/mod.rs

```diff
@@ -1,10 +1,305 @@
+                    .await?;
+                self.oracle_boosting_sink.commit().await?;
+                tracing::info!("Data set download complete");
+            }
+
+            // We don't want to shut down in the middle of downloading a data set, so we hold off until
+            // we are sleeping
+            #[rustfmt::skip]
+            tokio::select! {
+                biased;
+                _ = shutdown.clone() => {
+                    tracing::info!("DataSetDownloaderDaemon shutting down");
+                    break;
+                }
+                _ = tokio::time::sleep(poll_duration.to_std()?) => {
+                    continue;
+                }
+            }
+        }
+
+        Ok(())
+    }
+}
+
+#[derive(Copy, Clone, sqlx::Type)]
+#[sqlx(type_name = "data_set_type")]
+#[sqlx(rename_all = "lowercase")]
+pub enum DataSetType {
+    Urbanization,
+    Footfall,
+    Landtype,
+}
+
+impl DataSetType {
+    pub fn to_prefix(self) -> &'static str {
+        match self {
+            Self::Urbanization => "urbanization",
+            Self::Footfall => "footfall",
+            Self::Landtype => "landtype",
+        }
+    }
+}
+
+// Better name welcome
+pub struct CheckForNewDataSetDaemon {
+    pool: PgPool,
+    store: FileStore,
+    data_set: DataSetType,
+}
+
+impl ManagedTask for CheckForNewDataSetDaemon {
+    fn start_task(
+        self: Box<Self>,
+        shutdown: triggered::Listener,
+    ) -> futures::prelude::future::LocalBoxFuture<'static, anyhow::Result<()>> {
+        let handle = tokio::spawn(self.run(shutdown));
+        Box::pin(
+            handle
+                .map_err(anyhow::Error::from)
+                .and_then(|result| async move { result.map_err(anyhow::Error::from) }),
+        )
+    }
+}
+
+impl CheckForNewDataSetDaemon {
+    pub fn new(pool: PgPool, store: FileStore, data_set: DataSetType) -> Self {
+        Self {
+            pool,
+            data_set,
+            store,
+        }
+    }
+
+    pub async fn run(self, shutdown: triggered::Listener) -> anyhow::Result<()> {
+        // We should check for new data sets more often than we download them. 15 minutes seems fine.
+        let poll_duration = Duration::minutes(15);
+
+        let prefix = self.data_set.to_prefix();
+        let mut latest_file_date: Option<DateTime<Utc>> =
+            sqlx::query_scalar("SELECT time_to_use FROM data_sets ORDER BY time_to_use WHERE data_set = $1 DESC LIMIT 1")
+                .bind(self.data_set)
+            .fetch_optional(&self.pool)
+            .await?;
+
+        loop {
+            #[rustfmt::skip]
+            tokio::select! {
+                biased;
+                _ = shutdown.clone() => {
+                    tracing::info!("CheckForNewDataSetDaemon shutting down");
+                    break;
+                }
+                _ = tokio::time::sleep(poll_duration.to_std()?) => {
+                    let mut new_data_sets = self.store.list(prefix, latest_file_date, None);
```

Comment:
Might it be worth considering when a new dataset is generated and uploaded to S3 ( by whoever and whatever process performs that process ) that a `manifest` is included alongside the dataset.  That would avoid the need to poll S3 and list files in this way and allow a flow using our regular S3 fileinfo poller to be used and thus reusing existing conventions.  This daemon would consume the manifest and based on the manifest data trigger a download of the new dataset file

Reply:
Actually this is a great question.. no manifest needed, but why doesn't this just use the normal file poller setup?

Reply:
I was assuming this isnt using the normal file poller as the files are too large and we only want to be processing the latest.  Each file replaces what was before, unlike more of our ingested files which are additive...

Reply:
but that's just describing how you handle a new file when the poller sees it.. there's nothing additive about the poller itself

Reply:
The point i was making is that info poller will auto decode the file and as these things are large we may not want to necessarily do that for every file....

Reply:
ah yes.. true.. these hextrees are disk based and shouldn't be read/streamed